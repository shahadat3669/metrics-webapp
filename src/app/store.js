import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';

export const store = configureStore({
  reducer: {},
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
  devTools: process.env.NODE_ENV !== 'production',
});
