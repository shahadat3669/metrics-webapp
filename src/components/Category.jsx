import React from 'react';

const Category = () => <div>Category</div>;

export default Category;
