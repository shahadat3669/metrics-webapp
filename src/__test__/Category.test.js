import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Category from '../components/Category';
import store from '../app/store';

describe('Rockets', () => {
  test('renders Rockets Page without crashing', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Category />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree.toJSON).toMatchSnapshot();
  });

  test('renders Rockets Page without crashing', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Category />
        </BrowserRouter>
      </Provider>,
    );
    container.querySelector('.mainTitle');
    expect(container.firstChild).toMatchSnapshot();

    container.querySelector('.first_color');
    expect(container.firstChild).toMatchSnapshot();
  });
});
