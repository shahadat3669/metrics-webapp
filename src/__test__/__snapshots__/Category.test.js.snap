// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets renders Rockets Page without crashing 1`] = `undefined`;

exports[`Rockets renders Rockets Page without crashing 2`] = `
<div>
  <div
    class="swiper swiper-initialized swiper-horizontal mySwiper"
  >
    <div
      class="swiper-wrapper"
      style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
    >
      <div
        class="swiper-slide"
        data-swiper-slide-index="0"
      >
        <div
          class="category"
        >
          <div
            class="category__wrapper"
          >
            <div
              class="category__img-wrapper"
            >
              <img
                alt="europe"
                class="category__img"
                src="assets/europe.svg"
              />
            </div>
            <div
              class="category__content"
            >
              <div
                class="category__content-wrapper"
              >
                <h3
                  class="category__title"
                >
                  europe
                </h3>
                <button
                  aria-hidden="true"
                  class="category__button"
                  type="button"
                >
                  view
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="1"
      >
        <div
          class="category"
        >
          <div
            class="category__wrapper"
          >
            <div
              class="category__img-wrapper"
            >
              <img
                alt="asia"
                class="category__img"
                src="assets/asia.svg"
              />
            </div>
            <div
              class="category__content"
            >
              <div
                class="category__content-wrapper"
              >
                <h3
                  class="category__title"
                >
                  asia
                </h3>
                <button
                  aria-hidden="true"
                  class="category__button"
                  type="button"
                >
                  view
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="2"
      >
        <div
          class="category"
        >
          <div
            class="category__wrapper"
          >
            <div
              class="category__img-wrapper"
            >
              <img
                alt="africa"
                class="category__img"
                src="assets/africa.svg"
              />
            </div>
            <div
              class="category__content"
            >
              <div
                class="category__content-wrapper"
              >
                <h3
                  class="category__title"
                >
                  africa
                </h3>
                <button
                  aria-hidden="true"
                  class="category__button"
                  type="button"
                >
                  view
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="3"
      >
        <div
          class="category"
        >
          <div
            class="category__wrapper"
          >
            <div
              class="category__img-wrapper"
            >
              <img
                alt="oceania"
                class="category__img"
                src="assets/oceania.svg"
              />
            </div>
            <div
              class="category__content"
            >
              <div
                class="category__content-wrapper"
              >
                <h3
                  class="category__title"
                >
                  oceania
                </h3>
                <button
                  aria-hidden="true"
                  class="category__button"
                  type="button"
                >
                  view
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="4"
      >
        <div
          class="category"
        >
          <div
            class="category__wrapper"
          >
            <div
              class="category__img-wrapper"
            >
              <img
                alt="americas"
                class="category__img"
                src="assets/americas.svg"
              />
            </div>
            <div
              class="category__content"
            >
              <div
                class="category__content-wrapper"
              >
                <h3
                  class="category__title"
                >
                  americas
                </h3>
                <button
                  aria-hidden="true"
                  class="category__button"
                  type="button"
                >
                  view
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="swiper-pagination swiper-pagination-clickable swiper-pagination-bullets swiper-pagination-horizontal"
    >
      <span
        class="swiper-pagination-bullet swiper-pagination-bullet-active"
      />
      <span
        class="swiper-pagination-bullet"
      />
      <span
        class="swiper-pagination-bullet"
      />
      <span
        class="swiper-pagination-bullet"
      />
      <span
        class="swiper-pagination-bullet"
      />
    </div>
  </div>
</div>
`;

exports[`Rockets renders Rockets Page without crashing 3`] = `
<div>
  <div
    class="swiper swiper-initialized swiper-horizontal mySwiper"
  >
    <div
      class="swiper-wrapper"
      style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
    >
      <div
        class="swiper-slide"
        data-swiper-slide-index="0"
      >
        <div
          class="category"
        >
          <div
            class="category__wrapper"
          >
            <div
              class="category__img-wrapper"
            >
              <img
                alt="europe"
                class="category__img"
                src="assets/europe.svg"
              />
            </div>
            <div
              class="category__content"
            >
              <div
                class="category__content-wrapper"
              >
                <h3
                  class="category__title"
                >
                  europe
                </h3>
                <button
                  aria-hidden="true"
                  class="category__button"
                  type="button"
                >
                  view
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="1"
      >
        <div
          class="category"
        >
          <div
            class="category__wrapper"
          >
            <div
              class="category__img-wrapper"
            >
              <img
                alt="asia"
                class="category__img"
                src="assets/asia.svg"
              />
            </div>
            <div
              class="category__content"
            >
              <div
                class="category__content-wrapper"
              >
                <h3
                  class="category__title"
                >
                  asia
                </h3>
                <button
                  aria-hidden="true"
                  class="category__button"
                  type="button"
                >
                  view
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="2"
      >
        <div
          class="category"
        >
          <div
            class="category__wrapper"
          >
            <div
              class="category__img-wrapper"
            >
              <img
                alt="africa"
                class="category__img"
                src="assets/africa.svg"
              />
            </div>
            <div
              class="category__content"
            >
              <div
                class="category__content-wrapper"
              >
                <h3
                  class="category__title"
                >
                  africa
                </h3>
                <button
                  aria-hidden="true"
                  class="category__button"
                  type="button"
                >
                  view
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="3"
      >
        <div
          class="category"
        >
          <div
            class="category__wrapper"
          >
            <div
              class="category__img-wrapper"
            >
              <img
                alt="oceania"
                class="category__img"
                src="assets/oceania.svg"
              />
            </div>
            <div
              class="category__content"
            >
              <div
                class="category__content-wrapper"
              >
                <h3
                  class="category__title"
                >
                  oceania
                </h3>
                <button
                  aria-hidden="true"
                  class="category__button"
                  type="button"
                >
                  view
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="4"
      >
        <div
          class="category"
        >
          <div
            class="category__wrapper"
          >
            <div
              class="category__img-wrapper"
            >
              <img
                alt="americas"
                class="category__img"
                src="assets/americas.svg"
              />
            </div>
            <div
              class="category__content"
            >
              <div
                class="category__content-wrapper"
              >
                <h3
                  class="category__title"
                >
                  americas
                </h3>
                <button
                  aria-hidden="true"
                  class="category__button"
                  type="button"
                >
                  view
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="swiper-pagination swiper-pagination-clickable swiper-pagination-bullets swiper-pagination-horizontal"
    >
      <span
        class="swiper-pagination-bullet swiper-pagination-bullet-active"
      />
      <span
        class="swiper-pagination-bullet"
      />
      <span
        class="swiper-pagination-bullet"
      />
      <span
        class="swiper-pagination-bullet"
      />
      <span
        class="swiper-pagination-bullet"
      />
    </div>
  </div>
</div>
`;
